src/averageAges.js

```diff
@@ -55,7 +68,28 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person => (
+    people.find(mother => person.mother === mother.name))
+    && (onlyWithSon
+      ? person.sex === 'm'
+      : true)
+  );
+
+  const differenceAges = children.map(child => {
+    const mother = people.find(person => person.name === child.mother);
+
+    const differenceAge = child.born - mother.born;
+
+    return differenceAge;
```

Comment:
You can return `child.born - mother.born` here without creating variable `differenceAge` which you instantly return